src/priv_verif_token.ts

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache-2.0 license found in the LICENSE file or at https://opensource.org/licenses/Apache-2.0
+
+import {
+    Evaluation,
+    EvaluationRequest,
+    FinalizeData,
+    Oprf,
+    VOPRFClient,
+    VOPRFServer,
+    generateKeyPair,
+    type DLEQParams,
+    type Group,
+    type SuiteID,
+    type HashID,
+    DLEQProof,
+} from '@cloudflare/voprf-ts';
+
+import { joinAll } from './util.js';
+import {
+    AuthenticatorInput,
+    Token,
+    TokenChallenge,
+    type TokenTypeEntry,
+} from './auth_scheme/private_token.js';
+
+export interface VOPRFExtraParams {
+    suite: SuiteID;
+    group: Group;
+    Ne: number;
+    Ns: number;
+    Nk: number;
+    hash: HashID;
+    dleqParams: DLEQParams;
+}
+
+const VOPRF_SUITE = Oprf.Suite.P384_SHA384;
+const VOPRF_GROUP = Oprf.getGroup(VOPRF_SUITE);
+const VOPRF_HASH = Oprf.getHash(VOPRF_SUITE) as HashID;
+const VOPRF_EXTRA_PARAMS: VOPRFExtraParams = {
+    suite: VOPRF_SUITE,
+    group: VOPRF_GROUP,
+    Ne: VOPRF_GROUP.eltSize(),
+    Ns: VOPRF_GROUP.scalarSize(),
+    Nk: Oprf.getOprfSize(VOPRF_SUITE),
+    hash: VOPRF_HASH,
+    dleqParams: {
+        gg: VOPRF_GROUP,
+        hashID: VOPRF_HASH,
+        hash: Oprf.Crypto.hash,
+        dst: '',
```

Comment:
define `dst` as a separate constant to better identify it down the line. It might be worth having a recommended destination in cloudflare/voprf-ts, but I'm not sure there.

Reply:
It's not even actually needed to decode a  DLEQProof. A circular dependency (that I'm foggy on the ramifications of) triggered a refactoring to put the hash etc in the DLEParams. Really all you need is the group id for decoding the proof, as it is just two Scalar objects: {c, s}

But then again, you shouldn't even need to decode the proof at all here, except for the that the voprf-ts lib api accepts "wet" parameters (decoded VOPRF objects, not bytes). If the the EvalReq and Eval were already encode as objects, with all leaves as Uint8Array already, then at the application level (i.e. here in privacypass-ts) you could pick and choose from the fields and place them in app level packets.

Reply:
Ala: 

```typescript
export type DLEQProof = { c: Scalar; s: Scalar; serialize(): Uint8Array }
// OPRF protocol only specifies encodings for these elements
export type OPRFElement = Elt | Scalar | DLEQProof

// So we only encode elements/bytes, and leave packet encoding to application layer
export interface EvaluationRequest {
    readonly blinded: Array<Serialized<Elt>>
}

export interface BaseEvaluation {
    readonly mode: ModeID
    readonly evaluated: Array<Serialized<Elt>>
}

export interface VerifiableEvaluation extends BaseEvaluation {
    readonly proof: Serialized<DLEQProof>
}
```

Reply:
Regarding DST and all other params, those will be abstracted in the voprf-ts package. For now let's go with that, a refactor will solve that issue later.